src/lineup/LineUpRankingButtons.ts

```diff
@@ -2,51 +2,93 @@
  * Created by sam on 13.02.2017.
  */
 
-import {createStackDesc} from 'lineupjs/src/model';
+import {createStackDesc, IColumnDesc} from 'lineupjs/src/model';
 import * as d3 from 'd3';
 import {IDType, resolve} from 'phovea_core/src/idtype';
 import {IPlugin, IPluginDesc, list as listPlugins} from 'phovea_core/src/plugin';
 import {editDialog} from '../storage';
 import {EventHandler} from 'phovea_core/src/event';
+import {FormElementType, IFormElementDesc} from '../form/interfaces';
+import {OrdinoFormIds} from '../constants';
+import {IScoreLoader, IScoreLoaderExtensionDesc} from '../ScoreLoadingWrapper';
+import FormBuilder from '../form/FormBuilder';
+import {IButtonElementDesc} from '../form/internal/FormButton';
+import wrap from '../ScoreLoadingWrapper';
 import LineUp from 'lineupjs/src/lineup';
 
+interface IColumnWrapper<T> {
+  text: string;
+  action: (param: T) => void;
+  plugins?: T[];
+}
+
+interface IWrappedColumnDesc {
+  text: string;
+  id: string;
+  column: IColumnDesc;
+}
+
 export class LineUpRankingButtons extends EventHandler {
 
   static readonly SAVE_NAMED_SET = 'saveNamedSet';
   static readonly ADD_SCORE_COLUMN = 'addScoreColumn';
   static readonly ADD_TRACKED_SCORE_COLUMN = 'addTrackedScoreColumn';
 
+  private readonly $ul;
+
   constructor(private lineup: LineUp, private $node: d3.Selection<any>, private idType: IDType, private extraArgs: any) {
     super();
 
+    this.$ul = this.$node.append('ul').classed('ordino-button-group', true);
+
     this.appendDownload();
     this.appendSaveRanking();
     this.appendMoreColumns();
+    this.appendUpload();
+  }
+
+  private createMarkup(title: string = '', linkClass: string = '', linkListener: (param: any) => void | null, liClass: string = '') {
```

Comment:
does it make sense that all parameter are optional? especially since you almost always provide all of them.